paper-linear-progress.html

```diff
@@ -0,0 +1,897 @@
+         * representing the percentage complete of buffering. Note that the value will be clamped
+         * if it is less than/greater than 0/100, respectively.
+         */
+        bufferValue: {
+          type: Number,
+          observer: '_bufferValueChanged',
+          value: 100
+        },
+
+        /**
+         * When set, progress bars animate across the element, expressing an operation which takes
+         * an unknown amount of time.
+         */
+        indeterminate: {
+          type: Boolean,
+          observer: '_indeterminateChanged',
+          // We give a default value so that correct ARIA attributes are applied when element is
+          // initialized.
+          value: false
+        },
+
+        /**
+         * When set, will reverse all animations, transforms, and transitions occuring within the
+         * element. Useful when working with RTL languages, as well as distinguishing between phases
+         * of an operation.
+         */
+        reversed: {
+          type: Boolean,
+          observer: '_reversedChanged'
+        },
+
+        /**
+         * When set, will display buffering dots which appear to animate across the element. Useful
+         * when displaying progress for operations involving media, during which content may need to
+         * be buffered.
+         */
+        buffering: {
+          type: Boolean,
+          observer: '_bufferingChanged'
+        },
+
+        /**
+         * When set, the linear progress element will be hidden from view. Note that this is
+         * _different_ from `hidden`, as it does not set `display: none;` but rather just changes
+         * the element's opacity.
+         */
+        closed: {
+          type: Boolean,
+          observer: '_closedChanged'
+        },
+
+        /**
+         * Animation configuration for NeonAnimationRunnerBehavior.
+         */
+        animationConfig: {
+          value: function() {
+            // NOTE(traviskaufman): This is initialized within the attached() method, once we know
+            // the height of the progress indicator.
+            return {};
+          }
+        },
+
+        __needsBufferingAnimationOnAttached: {
+          // HACK: We want to have buffering play an animation when it's set, but we can't
+          // animate unless we first have the computed height of the element, which we can only
+          // retrieve once the element's attached. Therefore, we set up a queue of animations that
+          // need to be run, and if we encounter any animations. Note that this can be fixed once
+          // https://github.com/Polymer/polymer/issues/2399 is resolved. Then, we can do all
+          // animations within CSS.
+          type: Boolean,
+          value: false
+        }
+      },
+
+      hostAttributes: {
+        role: 'progressbar'
+      },
+
+      ready: function() {
+        this.toggleClass('isIE', /MSIE/.test(navigator.userAgent));
```

Comment:
No problem! It's a difficult feature to test because you essentially have to observe the behavior of an animating element before you can conclude whether or not the bug exists. Also, it seems like it would be at an external level trying to make assertions about internal aspects of a rendering engine.

That being said, given there are [animation events](http://www.sitepoint.com/css3-animation-javascript-event-handlers/) that one could in theory try to hook into `animationiteration` and attempt to check whether or not an element resets to the correct starting position. That might look something like:

* Instatiate some sentinel element with an animation as a test.
* Wait for the animation iteration event to be fired.
* When it's fired, check the "position" of the element within the document.

However, there are a few flaws in this logic:

* It requires the creation and performance of an animating element, separately from handling the animation for the actual element itself. Not to mention that feature testing this means delaying the actual instantiation/animation of the component.
* Whether or not the element will be in a state reflective of the initial keyframe block when the event is fired would have to be deterministic. I'm not sure this is the case.
* There would have to be a way of computing the element's transform within the animation, since the underlying CSS value would not change. [This blog post](http://www.kirupa.com/html5/get_the_value_of_css_properties_as_they_are_animating.htm) outlines a way to do this, but due to a `getComputedStyle` call within a `requestAnimationFrame` block it would induce massive performance overhead. And again, there would have to be some guarantee that you could observe the property at frame 0. Furthermore, what if the CSS property values reflect what _should_ be the state but the engine interprets it differently? How would we know this?

IMHO given that this is a vendor-specific bug, and that vendor explicitly makes clear in the bug report that they will not be fixing it, it seems like UA-sniffing is an appropriate approach.

As a completely alternative approach, we could perform the entire animation using JS/`requestAnimationFrame`. Or try stitching together transitions. For obvious reasons I'd argue against this :)